File: ui/src/auth/role.ts
import { GetUserAuthResponse } from '../client/http/default';

export enum Role {
  UNKNOWN = 'unknown',
  ADMIN = 'admin',
  MEMBER = 'member',
}

export const convertGetUserAuthRole = (role: GetUserAuthResponse['user_role']): Role => {
  switch (role) {
    case 'unknown':
      return Role.UNKNOWN;
    case 'super_admin':
      return Role.ADMIN;
    case 'admin':
      return Role.ADMIN;
    case 'member':
      return Role.MEMBER;
    default:
      return Role.UNKNOWN;
  }
};

File: permission/role.go
// Global package for user permission on the attendance system.
package permission

type Role string

const (
	RoleNotSpecified Role = ""
	RoleUnknown      Role = "unknown"
	// Super admin has the highest authority.
	// It can perform any APIs. It is for the developers or the one who has
	// the ownership of the attendance system.
	RoleSuperAdmin Role = "super_admin"
	// Admin is the second highest authority.
	// It can perform certain APIs such as creating a session, updating a session, etc.
	RoleAdmin Role = "admin"
	// Member is the lowest authority. They can only view certain data
	// or access their own data.
	RoleMember Role = "member"
)
